python-package/lightgbm/basic.py

```diff
@@ -1691,6 +1683,9 @@ def __init__(self, params=None, train_set=None, model_file=None, model_str=None,
             if not isinstance(train_set, Dataset):
                 raise TypeError('Training data should be Dataset instance, met {}'
                                 .format(type(train_set).__name__))
+            train_set.construct()
+            # copy the parameters from train_set
+            params.update(train_set.get_params())
```

Comment:
Can't 
```
updated_params = train_set.params
updated_params.update(params)
```
(make Booster's params more important than Dataset's ones) help to get rid of `get_params()` method?

Reply:
Indeed, this can save many codes